Add test file using the parameters from Mulyukova and Bercovici.

alarshi opened the pull request:
This PR adds the grain growth and reduction parameters using Mulyukova and Bercovici (2018) in the pinned state limit. Currently, the grain growth prefactor is not implemented from their paper because of unit conversion terms (\micro m to m), which was not straightforward.


* [x] I have read the guidelines in our [CONTRIBUTING.md](../blob/main/CONTRIBUTING.md) document.

### For all pull requests:

* [x] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).

Commit:
Add test file using the parameters from Mulyukova and Bercovici.

Commit:
Use constant temperature and only dislocation creep.

Commit:
use the right grain growth prefactor.

Commit:
Working test.

Add log and statistics for the full model run.

remove files associated with 200Ma model time.

jdannberg commented on tests/grain_size_strain_pinned_state.prm:
```diff
@@ -0,0 +1,52 @@
+
+    # Mulyukova and Bercovici (2018)
+    # Diffusion creep
+    set Diffusion creep prefactor            =1e-200 # s^-1 Pa^-1 m^p
```
Can you add a comment here that this is to make the diffusion viscosity so large that all deformation occurs by dislocation creep, which makes testing simpler.

jdannberg commented on tests/grain_size_strain_pinned_state.prm:
```diff
@@ -0,0 +1,52 @@
+
+include grain_size_strain.prm
+
+set Output directory                       = output_grain_size_strain_pinned_state
+
```
You can remove this.

jdannberg commented on tests/grain_size_strain_pinned_state.prm:
```diff
@@ -0,0 +1,52 @@
+# Over time the model maintains nearly constant temperatures, pressures and
+# strain rates. The grain size in the test approaches the equilibrium
+# grain size for t -> infinity. At 200 Myr the equilibrium grain size should
+# be 0.0165746 m.
```
Can you add a comment here that because we can not let the test run for such a long time, the test result is very different from that number?
And it would also be good to refer to the python notebook that computes the number.

jdannberg reviewed (commented):
Okay, great that this works! Can you attach the log file of the model you ran for 200 Myr just so that we have the exact result documented in case we ever need to compare to that?

alarshi commented:
Thank you for the review! I addressed your comments and added the log file for the full model run. I am not sure is the best way to include them, but for now, they are in the same folder.
This is the plot of grain size evolution for this test run, to compare the values.

![image](https://user-images.githubusercontent.com/19296817/220907968-eb2d42bb-702f-4772-b130-f0b1424b687b.png)

alarshi commented:
[statistics.txt](https://github.com/grain-damage-Mulyukova/aspect/files/11536243/statistics.txt)

Statistics file for this model when end time is 200Ma.

alarshi commented:
[statistics200Ma.txt](https://github.com/grain-damage-Mulyukova/aspect/files/11078737/statistics200Ma.txt)

Statistics file for this model when end time is 200Ma.

alarshi commented:
[log.zip](https://github.com/grain-damage-Mulyukova/aspect/files/11536251/log.zip)

Compressed log file for the model with end time of 200Ma.

alarshi commented:
I removed the larger files (model time of 200 Ma) and instead attached them here if we want to look at them.

alarshi commented:
@jdannberg : I have squashed the commits now. Thanks!
